test_conformance/integer_ops/test_integer_dot_product.cpp

```diff
@@ -0,0 +1,380 @@
+//
+// Copyright (c) 2021 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+// This is needed for std::numeric_limits<>::min() and max() to work on Windows.
+#if defined(_WIN32)
+#define NOMINMAX
```

Comment:
Yeah... un-defining `min` and `max` as done in `test_subgroups` is another possible solution.

A few other CTS tests add parentheses using the pattern '(std::min)' or `(std::numeric_limits<>::min)`, which also works around the problem.

I think a good long-term solution will be to put something in a shared header file so this will usually "just work".

Do you have a preference for this PR?  Happy to adjust to whatever the group decides.

Reply:
Not un-defining. I was requested to remove un-defining so I've added parentheses:
`(std::numeric_limits<Ty>::min)();`
`(std::numeric_limits<Ty>::max)();`

Reply:
Maybe we could add something to https://github.com/KhronosGroup/OpenCL-CTS/blob/master/test_common/harness/compat.h? If the problem exists in multiple places then I suggest we create a dedicated issue listing all of them and keep what was done here.